src/dmd/parse.d

```diff
@@ -2152,6 +2154,42 @@ final class Parser(AST) : Lexer
                             cppmangle = token.value == TOK.class_ ? CPPMANGLE.asClass : CPPMANGLE.asStruct;
                             nextToken();
                         }
+                        else if (token.value == TOK.string_) // extern(C++, "namespace", "namespaces")
+                        {
+                            ignoreCppSymbols = true;
+                            idents = new AST.Identifiers();
+
+                            while(1)
+                            {
+                                AST.StringExp stringExp = cast(AST.StringExp)parsePrimaryExp();
+                                const(char)[] name = stringExp.toStringz();
+                                if(name.length == 0)
+                                {
+                                    error("invalid zero length C++ namespace");
+                                    idents = null;
+                                    break;
+                                }
+                                else if(!Identifier.isValidIdentifier(name))
+                                {
+                                    error("C++ namespace `%s` is invalid", name.ptr);
```

Comment:
I think the rules for what is a valid identifier is well known. I changed the wording a bit to reflect that.

Reply:
There might be words that are keywords in C++ but not in D. How are those handled?

Reply:
Well it could depend on the C++ version, `constexpr` for example wasn't a keyword until C++11 same goes for `declytype`. You'd get a link error if you did try to use a C++ keyword, from the C++ side you would just receive an error. I don't know if it is worth maintaining that sort of list in D.